sycl/test/warnings/sycl-compiler-version-cl.cpp

```diff
@@ -0,0 +1,13 @@
+// RUN: %clangxx -fsycl -fsycl-host-compiler=cl %s \
+// RUN:     -fsycl-host-compiler-options="/std:c++17 /Zc:__cplusplus" -c \
+// RUN:     -o %t.out | FileCheck %s
```

Comment:
Do we need to emit an output file? If no, let's use `-fsyntax-only` like sycl/test/warnings/sycl-compiler-version-clang.cpp.

Reply:
`-fsycl-host-compiler` doesn't work with `-fsyntax-only` by design, so I have to deal with output files even if they aren't used